### PR TITLE
feat: add chunking by sentences; refactor code; remove semantic chunking for now

### PR DESCRIPTION
**Pull request by guenthermi**
- add chunking by sentences
- refactor code to move default values out of the chunker class
- remove semantic chunking for now

### CODE REVIEW

File: chunked_pooling/mteb_chunked_eval.py [hunk 1]
```diff
@@ -296,3 +300,9 @@ def _prune(queries, corpus, relevant_docs, prune_size):
                 new_corpus['test'][x] = corpus['test'][x]
             new_queries['test'][key] = queries['test'][key]
         return new_queries, new_corpus, new_relevant_docs
+
+    def _calculate_metrics_from_split(*args, **kwargs):
+        pass
+
+    def _evaluate_subset(*args, **kwargs):
```
Comment (violenil) [inline chunked_pooling/mteb_chunked_eval.py:307]: Where are `_calculate_metrics_from_split()` and `_evaluate_subset()` used?

Comment (violenil) [reply chunked_pooling/mteb_chunked_eval.py:307]: Or, what is the plan with them?

**Review by violenil — APPROVED** [review]
Good cleanup and refactoring. Let's merge ASAP 👍

File: chunked_pooling/mteb_chunked_eval.py (hunk 1 above)
Comment (guenthermi) [reply chunked_pooling/mteb_chunked_eval.py:307]: In some newer MTEB version the AbsTask class requires them to be implemented but as we overwrite the evaluation method without using them they are useless however cause errors when they are not implemented by the non-abstract classes

**Pull request merged**